### DISCUSSION TITLE
[V7] NumberInput has empty string value after field is cleared

### DISCUSSION THREAD

**Discussion by DJWalker42** [mantinedev/mantine#5177] [Q&A] [answered]
I am unsure if this is a bug. I have a number input that is controlled via a spread of form.getInputProps(). The initial value of the number input is null. If I enter a number into the input field then clear it, the value is now the empty string rather than being null. This affects my validate rules for my form where the field is required to have an actual numeric value i.e., not null but the empty string passes this check. For correct validation I must also check for the empty string, which requires '//ts-ignore' comments to suppress the error about comparing different types.

**Comment by rtivital** [comment, marked as answer]
`null` is not a valid NumberInput value – the component does not support this value type. Its value should be either a string or a number.

**Reply by DJWalker42** [reply]
I'll need to change my input values to be of type 'number | string' then. 

Actually, in making this switch it has also explained some weird behaviour I was seeing with form.isDirty(); because I was using an invalid initial state isDirty() was still returning true despite having cleared the field. The initial field was an invalid null but the cleared state is the valid empty string. Thanks, always useful to clear up confusion.
